test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -32,30 +32,127 @@
 /* Include paths for public enums, structures, and macros. */
 
 #include "mock_core_mqtt.h"
+#include "mock_core_mqtt_state.h"
+#include "mock_mqtt_agent.h"
+#include "mqtt_agent_command_functions.h"
 
 /**
- * @brief MQTT client identifier.
+ * @brief The agent messaging context.
  */
-#define MQTT_CLIENT_IDENTIFIER         "testclient"
+struct AgentMessageContext
+{
+    Command_t * pSentCommand;
+};
+
+/**
+ * @brief Command callback context.
+ */
+struct CommandContext
+{
+    MQTTStatus_t returnStatus;
+};
+
+/**
+ * @brief Message context to use for tests.
+ */
+static AgentMessageContext_t globalMessageContext;
+
+/**
+ * @brief Command struct pointer to return from mocked getCommand.
+ */
+static Command_t * pCommandToReturn;
+
+/**
+ * @brief Mock Counter variable to check callback is called on command completion.
+ */
+static uint32_t commandCompleteCallbackCount;
+
+/**
+ * @brief Mock Counter variable for calling stubReceive multiple times.
+ */
+static uint32_t receiveCounter;
+
+/* ========================================================================== */
+
+/**
+ * @brief A mocked send function to send commands to the agent.
+ */
+static bool stubSend( AgentMessageContext_t * pMsgCtx,
+                      const void * pData,
+                      uint32_t blockTimeMs )
+{
+    Command_t ** pCommandToSend = ( Command_t ** ) pData;
+
+    pMsgCtx->pSentCommand = *pCommandToSend;
+    return true;
+}
 
 /**
- * @brief A sample network context that we set to NULL.
+ * @brief A mocked receive function for the agent to receive commands.
  */
-#define MQTT_SAMPLE_NETWORK_CONTEXT    ( NULL )
+static bool stubReceive( AgentMessageContext_t * pMsgCtx,
+                         void * pBuffer,
+                         uint32_t blockTimeMs )
+{
+    Command_t ** pCommandToReceive = ( Command_t ** ) pBuffer;
+
+    if( receiveCounter == 0 )
+    {
+        *pCommandToReceive = pMsgCtx->pSentCommand;
+        receiveCounter++;
+        return true;
+    }
+    else
+    {
+        *pCommandToReceive = NULL;
+        receiveCounter++;
+        return false;
+    }
```

Comment:
Nit: This could fit in two lines if you wanted but it's not really that important.
```suggestion
    *pCommandToReceive = ( receiveCounter ) ? NULL : pMsgCtx->pSentCommand;
    return ( receiveCounter++ == 0 );
```